server/core/src/main/java/dev/slimevr/protocol/rpc/settings/RPCSettingsHandler.java

```diff
@@ -298,44 +298,36 @@ public void onChangeSettingsRequest(GenericConnection conn, RpcMessageHeader mes
 			}
 
 			if (ratios != null) {
-				if (ratios.hasImputeWaistFromChestHip()) {
-					hpm
-						.setValue(
-							SkeletonConfigValues.WAIST_FROM_CHEST_HIP_AVERAGING,
-							ratios.imputeWaistFromChestHip()
-						);
-				}
-				if (ratios.hasImputeWaistFromChestLegs()) {
-					hpm
-						.setValue(
-							SkeletonConfigValues.WAIST_FROM_CHEST_LEGS_AVERAGING,
-							ratios.imputeWaistFromChestLegs()
-						);
-				}
-				if (ratios.hasImputeHipFromChestLegs()) {
-					hpm
-						.setValue(
-							SkeletonConfigValues.HIP_FROM_CHEST_LEGS_AVERAGING,
-							ratios.imputeHipFromChestLegs()
-						);
-				}
-				if (ratios.hasImputeHipFromWaistLegs()) {
-					hpm
-						.setValue(
-							SkeletonConfigValues.HIP_FROM_WAIST_LEGS_AVERAGING,
-							ratios.imputeHipFromWaistLegs()
-						);
-				}
-				if (ratios.hasInterpHipLegs()) {
-					hpm.setValue(SkeletonConfigValues.HIP_LEGS_AVERAGING, ratios.interpHipLegs());
-				}
-				if (ratios.hasInterpKneeTrackerAnkle()) {
-					hpm
-						.setValue(
-							SkeletonConfigValues.KNEE_TRACKER_ANKLE_AVERAGING,
-							ratios.interpKneeTrackerAnkle()
-						);
-				}
+				hpm
+					.setValue(
+						SkeletonConfigValues.WAIST_FROM_CHEST_HIP_AVERAGING,
+						ratios.imputeWaistFromChestHip()
+					);
+				hpm
+					.setValue(
+						SkeletonConfigValues.WAIST_FROM_CHEST_LEGS_AVERAGING,
+						ratios.imputeWaistFromChestLegs()
+					);
+				hpm
+					.setValue(
+						SkeletonConfigValues.HIP_FROM_CHEST_LEGS_AVERAGING,
+						ratios.imputeHipFromChestLegs()
+					);
+				hpm
+					.setValue(
+						SkeletonConfigValues.HIP_FROM_WAIST_LEGS_AVERAGING,
+						ratios.imputeHipFromWaistLegs()
+					);
+				hpm
+					.setValue(
+						SkeletonConfigValues.HIP_LEGS_AVERAGING,
+						ratios.interpHipLegs()
+					);
+				hpm
+					.setValue(
+						SkeletonConfigValues.KNEE_TRACKER_ANKLE_AVERAGING,
+						ratios.interpKneeTrackerAnkle()
+					);
```

Comment:
wha? why are you removing that if it can be null

Reply:
It’s not null, it’s just a check if == 0

Reply:
It's a check if the pointer is 0, not the value. The "null" part is in the flatbuffers.

Reply:
It returns false if the value is 0, I tested it.
I tested changing other settings and it worked fine. If we want to keep the ‘has’ check, we’d have to hardcode ‘0’ to another value (like -1 or something) both in GUI and in server… Not convinced this is needed tho

Reply:
pls dont remove this, it returns if the pointer doesnt exist in flatbuffer :p
```java
  public boolean hasImputeWaistFromChestHip() { return 0 != __offset(4); }
  public float imputeWaistFromChestHip() { int o = __offset(4); return o != 0 ? bb.getFloat(o + bb_pos) : 0f; }
```

Reply:
What do you suggest we do then? use -1 to represent 0?

Reply:
just keep doing the has thing, if we used kotlin flatbuffers it would be simpler >~<

Reply:
I cannot keep doing the has thing, I **need** to know when the value is 0.

Reply:
imputeWaistFromChestHip will return 0 if its value is 0...

Reply:
Which is why I originally removed the has check, why are you telling me that?

Reply:
Fixed in https://github.com/SlimeVR/SlimeVR-Server/pull/743/commits/6ef30d7f6af4dfa28bd9693eec1398690e6fc977